### PR TITLE
feat(tablekit-core): add `data-variant` as prop for `MenuItem`

### PR DESCRIPTION
**Pull request by nkgardose**
## Changes
* Added `data-variant` prop to `MenuItem`

## Screenshot
Screenshot 2023-08-03 at 19 12 12

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/tablekit-core@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-css@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-react-css@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-react-datepicker@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-react-fit-content-textarea@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-react-select@2.1.0-canary.196.5758751898.0
  npm install @tablecheck/tablekit-react@2.1.0-canary.196.5758751898.0
  # or 
  yarn add @tablecheck/tablekit-core@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-css@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-react-css@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-react-datepicker@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-react-fit-content-textarea@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-react-select@2.1.0-canary.196.5758751898.0
  yarn add @tablecheck/tablekit-react@2.1.0-canary.196.5758751898.0
  ```
</details>

### CODE REVIEW

File: system/core/src/components/MenuItem.ts [hunk 1]
```diff
@@ -30,7 +42,8 @@ export const stateStylesObjects: Record<
   hover: {
     color: 'var(--text)',
     background: 'var(--surface-hover)'
-  }
+  },
+  ...variantStylesObjects
```
Comment (SimeonC) [inline system/core/src/components/MenuItem.ts:46]: Variants aren't stateStyles - these are essentially just for react-select integration.

File: system/core/src/components/MenuItem.ts [hunk 2]
```diff
@@ -53,8 +66,17 @@ export const baseStylesObject: CSSObject = {
 
 export interface Props {
   'data-selected'?: boolean;
+  'data-variant'?: boolean;
```
Comment (SimeonC) [inline system/core/src/components/MenuItem.ts:69]:
```suggestion
  'data-variant'?: (typeof variants)[number];
```

**Review by SimeonC — CHANGES_REQUESTED** [review]
Have a look at how variants are done in the [`Alert` component](https://github.com/tablecheck/tablekit/blob/6630a53a2bc68c01d5ae3c1c0e4385c9baeb6464/system/core/src/components/Alert.ts#L71) and follow that.

File: system/core/src/components/MenuItem.ts (hunk 1 above)
Comment (nkgardose) [reply system/core/src/components/MenuItem.ts:46]: I understand, thanks. I'll move it to base styles

Comment (nkgardose) [reply system/core/src/components/MenuItem.ts]: Done

File: system/core/src/components/MenuItem.ts (hunk 2 above)
Comment (nkgardose) [reply system/core/src/components/MenuItem.ts]: Done!

**Review by SimeonC — APPROVED** [review]

**Review by k-baldauf — APPROVED** [review]

**Pull request merged**